File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        k=k%n;
        k=n-k;

        vector<int>arr=nums;
        for(int i=0;i<n;i++){
            arr[i]=nums[(i+k)%n];
        }   
        nums=arr;
    
         }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        
        n=len(nums)
        k=k%n
        arr=[0]*n
        for i in range(n):
            arr[(i + k) % n] = nums[i]
            
            #reverse(arr.begin(),arr.begin()+(n-k))
        for i in range(0,n):
            nums[i]=arr[i]

        
        